src/lib/core/tests/TestCHIPTLV.cpp

```diff
@@ -211,14 +212,14 @@ void TestString(nlTestSuite * inSuite, TLVReader & reader, uint64_t tag, const c
     uint32_t expectedLen = strlen(expectedVal);
     NL_TEST_ASSERT(inSuite, reader.GetLength() == expectedLen);
 
-    char * val = static_cast<char *>(malloc(expectedLen + 1));
+    char * val = static_cast<char *>(chip::Platform::MemoryAlloc(expectedLen + 1));
```

Comment:
I would suggest using https://github.com/project-chip/connectedhomeip/blob/master/src/lib/support/ScopedBuffer.h in here if possible, to not need to pair up alloc and free.

Reply:
I'd prefer to do a 1:1 swap in this PR and all the improvements in a separate one, to keep the history clearer.